src/node_url.cc

```diff
@@ -125,6 +106,71 @@ enum url_error_cb_args {
 #undef XX
 };
 
+#define CHAR_TEMPLATE(T, n)                                                   \
+template <typename T,                                                       \
+          typename = typename std::enable_if<std::is_integral<T>::value &&  \
+                                             sizeof(T) >= n / 8, T>::type>
+
```

Comment:
Tbh this does feel a bit like overkill ;) Plain `template <typename T> static inline`s should do the job, right?